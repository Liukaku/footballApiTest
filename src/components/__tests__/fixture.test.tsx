import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Fixture from "../fixture";

const mockFixture = {
  id: 1,
  referee: "Boss Man",
  timezone: "string",
  date: "string",
  timestamp: 1,
  periods: {
    first: 1,
    second: 2,
  },
  venue: {
    id: 2,
    name: "string",
    city: "string",
  },
  status: {
    long: "Full Time",
    short: "FT",
    elapsed: 90,
  },
};

const mockTeams = {
  home: {
    id: 1,
    name: "Home Team",
    logo: "https://media.api-sports.io/football/teams/1.png",
    winner: true,
  },
  away: {
    id: 2,
    name: "Away Team",
    logo: "https://media.api-sports.io/football/teams/1.png",
    winner: false,
  },
};

const mockGoals = {
  home: 2,
  away: 0,
};

describe("Fixture", () => {
  it("check component matches snapshot", () => {
    const fixture = renderer
      .create(
        <Fixture fixture={mockFixture} teams={mockTeams} goals={mockGoals} />
      )
      .toJSON();

    expect(fixture).toMatchSnapshot();
  });
});
