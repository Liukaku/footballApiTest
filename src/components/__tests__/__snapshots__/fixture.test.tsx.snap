// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixture returns true 1`] = `
<div
  className=" flex flex-wrap h-full items-center overflow-hidden"
>
  <img
    alt=""
    className="opacity-50 md:left-0 left-16 absolute blur-sm z-0 aspect-square md:w-auto w-1/4"
    data-nimg="1"
    decoding="async"
    height={150}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=https%3A%2F%2Fmedia.api-sports.io%2Ffootball%2Fteams%2F1.png&w=384&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Fmedia.api-sports.io%2Ffootball%2Fteams%2F1.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.api-sports.io%2Ffootball%2Fteams%2F1.png&w=384&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={150}
  />
  <div
    className="w-full text-center z-20 font-bold"
  >
    <h1
      className=""
    >
      Home Team
       vs 
      Away Team
    </h1>
    <h3>
      2
       - 
      0
    </h3>
    <h3>
      Full Time
    </h3>
  </div>
  <img
    alt=""
    className="opacity-50 absolute blur-sm z-0 md:right-0 right-16 aspect-square md:w-auto w-1/4"
    data-nimg="1"
    decoding="async"
    height={150}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=https%3A%2F%2Fmedia.api-sports.io%2Ffootball%2Fteams%2F1.png&w=384&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Fmedia.api-sports.io%2Ffootball%2Fteams%2F1.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.api-sports.io%2Ffootball%2Fteams%2F1.png&w=384&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={150}
  />
</div>
`;
